libraries/BLE/src/BLEDevice.cpp

```diff
@@ -629,10 +629,15 @@ void BLEDevice::addPeerDevice(void* peer, bool _client, uint16_t conn_id) {
 	m_connectedClientsMap.insert(std::pair<uint16_t, conn_status_t>(conn_id, status));
```

Comment:
shouldn't the same mux be used here and any place where `m_connectedClientsMap` is accessed?

Reply:
As far as I'm noticed, the `insert` and other operations to `m_connectedClientsMap` will not be invoked concurrently. So, it should not be necessary🤔.

Reply:
are you positive that insert and find/erase can not happen at the same time?

Reply:
`erase` operation will happen at the same time above the situation of this pull request. `find` operation will not cause concurrent problem. Regretfully, I haven't found enough evidence so far to promise if the `insert` operation will be invoked simultaneously in this library, however this is unlikely to happen. Therefore, I only made the `BLEDevice::removePeerDevice` serializable.